tracer/src/Datadog.Trace/DataStreamsMonitoring/Aggregation/DataStreamsMessagePackFormatter.cs

```diff
@@ -16,6 +16,12 @@ namespace Datadog.Trace.DataStreamsMonitoring.Aggregation
 {
     internal class DataStreamsMessagePackFormatter
     {
+        // extend the list as needed
+        private const int ApmProduct = 1; // 00000001
+        private const int DsmProduct = 2; // 00000010
+        private const int DjmProduct = 4; // 00000100, not used in .NET, but needs to be in sync with other languages, hence the unused constant
+        private const int ProfilingProduct = 8; // 00001000
```

Comment:
Small C# style comment: this would normally be done as an enum with `[Flags]`. Something like

```csharp
    [Flags]
    internal enum Products
    {
        // extend the list as needed
        Apm = 1,       // 00000001
        Dsm = 2,       // 00000010
        Djm = 4,       // 00000100, not used in .NET, but needs to be in sync with other languages, hence the unused constant
        Profiling = 8, // 00001000
    }

    internal class DataStreamsMessagePackFormatter
    {
```

Reply:
Or another option (a personal preference):

```csharp
internal class DataStreamsMessagePackFormatter
{
    // note that the enum can still be private and nested to encapsulate it 
    [Flags]
    private enum Products
    {
        None = 0, // This should be explicit IMO for Flags enums
        Apm = 1,            // 00000001
        Dsm = 1 << 1,       // 00000010
        Djm = 1 << 2,       // 00000100
        Profiling = 1 << 3, // 00001000
    }
}
```

Note also you can use separators in the binary version:

```csharp
Apm = 0b0000_0001
Dsm = 0b0000_0010
Djm = 0b0000_0100
```